### ISSUE TITLE
Price elasticity and small sample sizes

### ISSUE THREAD

**Issue by nfritz6** [microsoft/EconML#574]
Hello and congratulations for this great package!

Currently I am working on a project which is related to price elasticity of demand. So, I have no X variable, and the only thing I would like to know is the price elasticity. If I have understood correctly, by taking the logs of Y(quantity) and T(price), the constant elasticity can be given by the CATE intercept via using the const_marginal_effect. But I am not sure if my approach is correct.

Another issue that I have, is that each time I run my model, I obtain largely different results, due to the small sample size (e.g. 800 rows). In order to tackle this issue, I have tried to use a repeated cross validation (sklearn) with the cv parameter of SparseLinearDML, but it produces an error. My first models are tree based. Can you propose a different approach, or give me any advice on how to tackle this issue?

Thanks a lot!

**Comment by kbattocchi** [comment]
You are correct that the elasticity can be calculated as the effect of log(T) on log(Y).  I'm curious what you mean when you say that Y has a binomial distribution.

As to some other aspects of your question, it would be helpful if you could post the full stack trace for the error you're getting when you try to use cv.  In any case, one way to reduce the variability of the output is to use the `mc_iters` argument of the DML initializer to average multiple iterations of the estimation procedure, which should average out the noise due to the randomness inherent to our sample splitting.  Finally, if you do not have any Xs, then I don't think it makes sense to use SparseLinearDML instead of just regular LinearDML since the final stage regression will not be sparse.

**Comment by nfritz6** [comment]
Thank you very much for your reply!

Regarding the binomial distribution, most probably I used the wrong term. What I mean is that the relationship between T(price) and Y(demand) looks like this.
![output](https://user-images.githubusercontent.com/72734947/154228099-6bfbf3dd-ff34-4507-8b9d-2dccc5c8695c.png)

The above plot was created from the data of the Customer Segmentation: Estimate Individualized Responses to Incentives. Mine looks exactly like this, except that I have only two price levels. Therefore, in this case, are we able to use a LinearDML with polynomials or the approach should be different (e.g. another form of regression as a final model)?

As for what I tried with cv and by having in mind that the same can be done in other cases, like in the cv parameter of GridSearchCV, is the following:
![example](https://user-images.githubusercontent.com/72734947/154229496-9f122c4a-eefd-48c4-a2cc-e927b5c98e93.png)

**Comment by nfritz6** [comment]
And now that I fully understood the difference between X and W features, of course and I have Xs! Therefore, I am able to use non parametric models like the CausalForestsDML. Nevertheless, in my case, the results (coefficient - elasticity and p-value) of a SparseLinealDML with polynomials and a CausalForestDML, do not differ much.

I have two other questions though, which are more general:

1) Given the nature of the DML approach, do we need to worry about multicollinearity between covariates (Xs) and/or Confounders (Ws) when using regressors?

2) Is there a way to identify the impact and/or feature importance of Ws?

Finally, if the way that I tried to use the sklearn Repeated KFold with the CV parameter of the DML, makes sense to you, please let me know in order to post the full trace for the error.

**Comment by nfritz6** [comment]
Just a note. The time-series cross-validator works like a charm!

![tscv](https://user-images.githubusercontent.com/72734947/155968074-b6400989-6e7b-4f6c-90b2-93989ff02512.png)
